src/jaxfit/types.py

```diff
@@ -0,0 +1,31 @@
+from typing import Callable, Set, Tuple
+
+from jax.interpreters.partial_eval import DynamicJaxprTracer
```

Comment:
Is this like a "any jax thing" type? Because most functions should also accept concrete jax arrays in addition to tracers

Reply:
Indeed it's just the tracer - right now tests only ever use this type since it's always through the jit.

I'll make a Union type for this in types.py and we can just add stuff to it as the tool matures.